notebook-controller/tests/test_rock.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Tests for required artifacts to be present in the ROCK image.
+#
+
+from charmed_kubeflow_chisme.rock import CheckRock
+from pathlib import Path
+
+import os
+import logging
+import random
+import pytest
+import string
+import subprocess
+import yaml
+from pytest_operator.plugin import OpsTest
+
+@pytest.fixture()
+def rock_test_env():
+    """Yields a temporary directory and random docker container name, then cleans them up after."""
+    container_name = "".join([str(i) for i in random.choices(string.ascii_lowercase, k=8)])
+    yield container_name
+
+    try:
+        subprocess.run(["docker", "rm", container_name])
+    except Exception:
+        pass
+
+@pytest.mark.abort_on_fail
+def test_rock(ops_test: OpsTest, rock_test_env):
+    """Test rock."""
+    check_rock = CheckRock("rockcraft.yaml")
+    container_name = rock_test_env
+    LOCAL_ROCK_IMAGE = f"{check_rock.get_image_name()}:{check_rock.get_version()}"
+
+    # verify that all artifacts are in correct locations
+    subprocess.run(["docker", "run", LOCAL_ROCK_IMAGE, "exec", "ls", "-la", "/manager"], check=True)
```

Comment:
Similar to my previous comment, I think you want to check the stderr, stdout to ensure everything's running as expected. You may even want to check for [this](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) exception since you have `check=True`.

Reply:
Will add exception handling, but in this case it is just trivial check for presence of binary. It does not start service or binary at this point, just verifies that it is present in the ROCK.